icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -216,6 +293,35 @@ UBool U_CALLCONV cleanupUnitExtras() {
 
 void U_CALLCONV initUnitExtras(UErrorCode& status) {
     ucln_i18n_registerCleanup(UCLN_I18N_UNIT_EXTRAS, cleanupUnitExtras);
+    LocalUResourceBundlePointer unitsBundle(ures_openDirect(nullptr, "units", &status));
+
+    // Collect unitQuantities information into gSerializedUnitQuantitiesTrie and gQuantities.
```

Comment:
Each simple unit is backed by a numeric ID into `gSimpleUnits`.  Instead of making a BytesTrie for lookup, can't you just make an array with the same length as `gSimpleUnits`?

Reply:
I've done something a bit like this now, adding a map from "simple unit ID index" to "unitQuantities index". However we still need to look up strings for other use cases. I'm thus keeping the BytesTrie, using it to construct this new int->int map, as well as for the other use case (`U_I18N_API const char *getUnitQuantity(StringPiece baseUnitIdentifier, UErrorCode &status);`).

(We need to be able to look up derived units, like kilogram-meter-per-square-second [newton], so unitQuantities is about more than just simple units.)
- We need to map from the index of "newton" in gSimpleUnits to the